src/main/java/com/hivemq/extensions/services/auth/SecurityRegistryImpl.java

```diff
@@ -60,11 +71,15 @@ public void setEnhancedAuthenticatorProvider(
 
         checkNotNull(enhancedAuthenticatorProvider, "enhancedAuthenticatorProvider must not be null");
 
-        final IsolatedPluginClassloader classLoader =
-                (IsolatedPluginClassloader) enhancedAuthenticatorProvider.getClass().getClassLoader();
+        final HiveMQExtension extension =
+                hiveMQExtensions.getExtensionForClassloader(enhancedAuthenticatorProvider.getClass().getClassLoader());
+
+        if(extension == null || extension.getExtensionClassloader() == null){
+            return;
```

Comment:
Shouldn't this throw an IllegalStateException?

Reply:
no as we just dont care if an extension is disabled.